### PR TITLE
 Add a warning for the FileType function

### PR DESCRIPTION
**Pull request by vladmos**

### CODE REVIEW

File: warn/warn_bazel.go [hunk 1]
```diff
@@ -260,6 +260,48 @@ func ctxActionsWarning(f *build.File, fix bool) []*Finding {
...
+								arg = binary.Y
+							}
+							toReplace[*e] = arg
+							return
+						}
+					}
+					start, end := call.Span()
+					findings = append(findings,
+						makeFinding(f, start, end, "filetype",
+							"The FileType function is deprecated, use lists instead.", true, nil))
```
Comment (laurentlb) [inline warn/warn_bazel.go]: Unfortunately, it's not the only use-case. The other use-case is FileType.filter (https://docs.bazel.build/versions/master/skylark/lib/FileType.html#filter).
Updating the code to use lists is not always working. Maybe we should be more careful and avoid the automated fix?

File: warn/warn_bazel.go [hunk 2]
```diff
@@ -260,6 +260,48 @@ func ctxActionsWarning(f *build.File, fix bool) []*Finding {
 	return findings
 }
 
+func fileTypeWarning(f *build.File, fix bool) []*Finding {
+	findings := []*Finding{}
+	toReplace := map[build.Expr]build.Expr{}
+
+	var walk func(e *build.Expr, env *bzlenv.Environment)
+	walk = func(e *build.Expr, env *bzlenv.Environment) {
```
[the remark this answers is not in the document]
Comment (vladmos) [reply warn/warn_bazel.go]: Done.

File: WARNINGS.md [hunk 3]
```diff
@@ -275,3 +275,12 @@ instead.
...
 instead.
+
+--------------------------------------------------------------------------------
+
+## <a name="filetype"></a>The `FileType` function is deprecated
+
+The function `FileType` is deprecated. The main use-case was as an argument to
+the [`rule` function](https://docs.bazel.build/versions/master/skylark/lib/globals.html#rule).
+It's no longer needed, you can simply pass a list of strings to restrict the
+file types the rule accepts.
```
[the remark this answers is not in the document]
Comment (vladmos) [reply WARNINGS.md]: Done.

File: warn/warn_bazel.go (hunk 1 above)
Comment (vladmos) [reply warn/warn_bazel.go]: Removed the fix functionality, just a warning now.

**Pull request merged**